examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -282,8 +282,7 @@ CHIP_ERROR OTAProviderExample::SendQueryImageResponse(chip::app::CommandHandler
         response.metadataForRequestor.Emplace(chip::ByteSpan());
     }
 
-    ReturnErrorOnFailure(commandObj->AddResponseData(commandPath, response));
-
+    VerifyOrReturnError(commandObj->AddResponse(commandPath, response) == CHIP_NO_ERROR, EMBER_ZCL_STATUS_FAILURE);
```

Comment:
So what happens in practice is that the caller of this method will see the error and try to AddStatus on the commandObj.  Which will presumably just fail again, right?

Would it make sense to just have AddResponse return void to avoid that problem?  If we really feel like we need to expose the fact that AddResponse failed, we should audit all AddResponse callers to make sure they don't try to AddStatus after an AddResponse failure.

A followup for addressing this, by either audit or changing the signature, is best, as long as we make sure it happens... and we would need to very carefully review any future AddResponse code if we keep the current signature.

Reply:
Made it a void function now, however, I also replaced return to LogErrorOnFailure since the users are also logging the error codes.